### DISCUSSION TITLE
Two panels, wildly different proximity sensor behaviour

### DISCUSSION THREAD

**Discussion by fcassata** [seaky/nspanel_pro_tools_apk#111] [Q&A] [answered]
Hi all, I have two panels, one running FW 1.7, the other 2.2. The 1.7 has a usable proximity sensor, reporting 0 from a distance of about 2.5 meters, so I have no problems with waking the display when a person approaches. The one running 2.2 doesn't go lower than ~1000, and doesn't pick up on motion until you're about 20cm away, which makes the wake up on approach very unintuitive for people that don't know the inner workins of the sensor settings. Verbose log from the FW2.2 device in first comment, but nothing obvious there, just getting triggered...

My first question is, does anyone know if this is a firmware issue, or a hardware issue? Any other ideas or troubleshooting tips?

See screenshot from each one, with no one anywhere close to them.
![1](https://github.com/seaky/nspanel_pro_tools_apk/assets/97367937/5fc0f9ab-2798-4dfb-ab15-2e5e1e1f2935)
![screencap](https://github.com/seaky/nspanel_pro_tools_apk/assets/97367937/d2dedb79-62bd-48e8-952f-5c016b1cdaca)

**Comment by fcassata** [comment]
Log from FW 2.2 after enabling DEBUG+VERBOSE

06-30 19:48:29.781  1354  1354 I c       : Log level is DEBUG
06-30 19:48:30.812  1354  1354 I c       : Log level is VERBOSE
06-30 19:48:32.460  1354  1354 I c       : Log level is VERBOSE
06-30 19:48:33.689  1354  1354 I zygote64: Deoptimizing java.lang.Object s.l.put(java.lang.Object, java.lang.Object) due to JIT same target
06-30 19:48:33.724  1354  1354 I zygote64: Deoptimizing void g0.w.f(android.content.Context, g0.t0) due to JIT inline cache
06-30 19:48:33.725  1354  1354 I zygote64: Deoptimizing android.graphics.Typeface g0.t0.g(int, int, g0.w$a) due to JIT inline cache
06-30 19:48:33.789  1354  1359 I zygote64: Do full code cache collection, code=507KB, data=314KB
06-30 19:48:33.790  1354  1359 I zygote64: After code cache collection, code=489KB, data=285KB
06-30 19:48:33.848  1354  1359 I zygote64: Do partial code cache collection, code=493KB, data=294KB
06-30 19:48:33.849  1354  1359 I zygote64: After code cache collection, code=493KB, data=294KB
06-30 19:48:33.849  1354  1359 I zygote64: Increasing code cache capacity to 2MB
06-30 19:48:33.895  1354  1363 E PowerManager: WakeLock finalized while still held: NSPanelTools:PreventScreenOffLock
06-30 19:48:34.069  1354  1354 I zygote64: Deoptimizing void android.support.v4.widget.NestedScrollView.measureChildWithMargins(android.view.View, int, int, int, int) due to JIT inline cache
06-30 19:48:37.035  1354  1354 D MainActivity: BackStack:1
06-30 19:48:39.298  1354  1354 D MainActivity: BackStack:0
06-30 19:48:39.340  1354  1354 I c       : sensor registered (trigger threshold:1100)
06-30 19:48:39.346  1354  1354 I a       : light sensor registered (trigger below:1000,trigger above:2000)
06-30 19:48:39.348  1354  1354 I m       : Proximity sensor live view started
06-30 19:48:45.155  1354  1354 I zygote64: Deoptimizing void s0.c.onSensorChanged(android.hardware.SensorEvent) due to JIT inline cache
06-30 19:48:45.156  1354  1354 D c       : V1 sensor triggered 1719748125154
06-30 19:48:45.160  1354  1354 I d       : k_b has been received
06-30 19:48:45.162  1354  1354 D f       : event received
06-30 19:48:45.166  1354  1354 D c       : V1 sensor triggered 1719748125165
06-30 19:48:45.577  1354  1354 D c       : V1 sensor triggered 1719748125576
06-30 19:48:45.581  1354  1354 D c       : V1 sensor triggered 1719748125580
06-30 19:48:45.586  1354  1354 I d       : k_b has been received
06-30 19:48:45.589  1354  1354 D f       : event received
06-30 19:48:46.004  1354  1354 D c       : V1 sensor triggered 1719748126003
06-30 19:48:46.009  1354  1354 I d       : k_b has been received
06-30 19:48:46.012  1354  1354 D f       : event received
06-30 19:48:46.016  1354  1354 D c       : V1 sensor triggered 1719748126015
06-30 19:48:46.403  1354  1354 D c       : V1 sensor triggered 1719748126402
06-30 19:48:46.408  1354  1354 D c       : V1 sensor triggered 1719748126407
06-30 19:48:46.413  1354  1354 I d       : k_b has been received
06-30 19:48:46.416  1354  1354 D f       : event received
06-30 19:48:48.218  1354  1354 D MainActivity: BackStack:0
06-30 19:48:48.259  1354  1354 I a       : unregister
06-30 19:48:48.261  1354  1354 I m       : Proximity sensor live view stopped
06-30 19:48:51.397  1354  1354 I zygote64: Deoptimizing void g0.w.f(android.content.Context, g0.t0) due to JIT inline cache
06-30 19:48:51.398  1354  1354 I zygote64: Deoptimizing android.graphics.Typeface g0.t0.g(int, int, g0.w$a) due to JIT inline cache
06-30 19:48:55.024  1354  1354 D MainActivity: BackStack:1
06-30 19:48:59.284  1354  1354 D MainActivity: BackStack:0
06-30 19:48:59.327  1354  1354 I c       : sensor registered (trigger threshold:1100)
06-30 19:48:59.332  1354  1354 I a       : light sensor registered (trigger below:1000,trigger above:2000)
06-30 19:48:59.333  1354  1354 I m       : Proximity sensor live view started
06-30 19:49:01.014  1354  1354 D MainActivity: BackStack:0
06-30 19:49:01.053  1354  1354 I a       : unregister
06-30 19:49:01.055  1354  1354 I m       : Proximity sensor live view stopped
06-30 19:49:10.550  1354  1354 D MainActivity: BackStack:0
06-30 19:49:12.047  1354  1354 D MainActivity: BackStack:0
06-30 19:49:12.099  1354  1354 I c       : sensor registered (trigger threshold:1100)
06-30 19:49:12.105  1354  1354 I a       : light sensor registered (trigger below:1000,trigger above:2000)
06-30 19:49:12.106  1354  1354 I m       : Proximity sensor live view started
06-30 19:49:14.127  1354  1354 D MainActivity: BackStack:0
06-30 19:49:14.167  1354  1354 I a       : unregister
06-30 19:49:14.169  1354  1354 I m       : Proximity sensor live view stopped
06-30 19:49:15.891  1354  1354 D MainActivity: BackStack:0
06-30 19:49:15.931  1354  1354 I c       : sensor registered (trigger threshold:1100)
06-30 19:49:15.936  1354  1354 I a       : light sensor registered (trigger below:1000,trigger above:2000)
06-30 19:49:15.938  1354  1354 I m       : Proximity sensor live view started
06-30 19:49:16.424  1354  1354 D MainActivity: BackStack:0
06-30 19:49:16.475  1354  1354 I a       : unregister
06-30 19:49:16.476  1354  1354 I m       : Proximity sensor live view stopped
06-30 19:49:18.120  1354  1354 D MainActivity: onPause
06-30 19:49:18.361  1354  1400 W libEGL  : EGLNativeWindowType 0x73ef91b010 disconnect failed
06-30 19:49:18.361  1354  1400 D OpenGLRenderer: endAllActiveAnimators on 0x73ee7f3000 (RippleDrawable) with handle 0x73ef26d100
06-30 19:49:18.381  1354  1354 D MainActivity: onStop
06-30 19:49:24.322  1354  1354 D AlarmPingSender: Sending Ping at:1719748164322
06-30 19:49:24.326  1354  1354 D AlarmPingSender: Schedule next alarm at 1719748224326
06-30 19:49:24.326  1354  1354 D AlarmPingSender: Alarm scheule using setExactAndAllowWhileIdle, next: 60000

**Comment by seaky** [comment, marked as answer]
Sorry to say that but this sensors are not designed to sense meter distances. Originally the reliable sense distance is only couple of cm. Usually <20cm. So you are quite lucky to have one which has 2.5m long distance sensitivity.

**Reply by fcassata** [reply]
Thanks for the answer. At least I know it's not defective then, I'll hook them up to motion sensors to wake them up through MQTT.

And thanks for the app! These panels would be useless to me without it.

**Reply by Vladx71** [reply]
can you please share how to do it? I also have a new nspanel120 and wake-up by wave/proximity not working at all (from any distance), so since I have an mqtt server as well, it seems a great idea to wake up by some automation.

Thank you
L:

**Reply by fcassata** [reply]
> can you please share how to do it? I also have a new nspanel120 and wake-up by wave/proximity not working at all (from any distance), so since I have an mqtt server as well, it seems a great idea to wake up by some automation.
> 
> Thank you L:

Ok so if you have them connected to HA with NSPanel Pro Tools, you get a few button entities. To wake you just have to do a button.press action on button.yournspanel_wakeup and _sleep. You can also skip the whole MQTT thing and set up an integration called "Android Debug Bridge", that gives you access to sending ADB command directly to the OS on the panel, which can in some cases be more reliable.

I actually use a mix of these, I use the button.nspanel_wakeup to wake the screen, but I use the ADB command "adb shell input keyevent KEYCODE_POWER" to put it to sleep.

This is the action call for each:

```
target:
  entity_id: media_player.adb_nspanel_eldhus (the ADB integration creates "media player" devices for the panels
data:
  command: adb shell input keyevent KEYCODE_POWER
action: androidtv.adb_command

target:
  entity_id: button.nspanel_stofa_wakeup
data: {}
action: button.press
```

Let me know if you have any questions!

**Reply by seaky** [reply]
> > can you please share how to do it? I also have a new nspanel120 and wake-up by wave/proximity not working at all (from any distance), so since I have an mqtt server as well, it seems a great idea to wake up by some automation.
> 
> > 
> 
> > Thank you L:
> 
> 
> 
> Ok so if you have them connected to HA with NSPanel Pro Tools, you get a few button entities. To wake you just have to do a button.press action on button.yournspanel_wakeup and _sleep. You can also skip the whole MQTT thing and set up an integration called "Android Debug Bridge", that gives you access to sending ADB command directly to the OS on the panel, which can in some cases be more reliable.
> 
> 
> 
> I actually use a mix of these, I use the button.nspanel_wakeup to wake the screen, but I use the ADB command "adb shell input keyevent KEYCODE_POWER" to put it to sleep.
> 
> 
> 
> This is the action call for each:
> 
> 
> 
> ```
> 
> target:
> 
>   entity_id: media_player.adb_nspanel_eldhus (the ADB integration creates "media player" devices for the panels
> 
> data:
> 
>   command: adb shell input keyevent KEYCODE_POWER
> 
> action: androidtv.adb_command
> 
> 
> 
> target:
> 
>   entity_id: button.nspanel_stofa_wakeup
> 
> data: {}
> 
> action: button.press
> 
> ```
> 
> 
> 
> Let me know if you have any questions!

Why dont you use mqtt sleep command?

**Reply by fcassata** [reply]
Great question, I think the original reason was that some combination of nspanel pro tools settings and fullykiosk settings was making it work inconsistently. I'm pretty sure the issue was on my end so I'll test going back to the mqtt command to simplify...

**Reply by seaky** [reply]
> reat idea to wake up by some automation.

what do you mean "does not work from any distance"? what do you see on sensor tab? Does it stay 0 always?

**Reply by Vladx71** [reply]
wakeup is working, no I have only issue with triggering sleep if noone nearby as I wrote in the another thread

**Comment by Vladx71** [comment]
Thank you. The 'buttoon' to sleep is working fine and with the proximity
sensor I can wake up the device. However I have one issue, I can't find an
entity which can be used for trigger to turn off the display when there is
no proximity

I have only this
event.nspanel1_proximity

However, the events are quite interesting :)

[image: image.png]

**Reply by Vladx71** [reply]
This is the picture I mentioned above... so what should I use as a trigger?

![Screenshot 2024-09-30 at 09 47 39](https://github.com/user-attachments/assets/fc8b3606-1e04-4383-9a56-2ad68b891ce0)

**Reply by fcassata** [reply]
I have no idea what I'm looking at there or what those values are, you have to paste something else. Are you using the proximity sensor on the panel? What are those values?

**Reply by Vladx71** [reply]
This is the only entity for proximity sensor for this nspanel and the picture above represents the 'states' I see in MQTT

event.nspanel1_proximity

**Reply by fcassata** [reply]
The history shows it has been getting a lot of different values, what are those values?

**Comment by Vladx71** [comment]
I have no idea.

All I can see is ' detected an event' in different colors which if
beautiful but not very useful :)

L:
